src/v/model/record_batch_reader.h

```diff
@@ -280,6 +280,36 @@ make_memory_record_batch_reader(model::record_batch b) {
     return make_memory_record_batch_reader(std::move(batches));
 }
 
+// clang-format off
+template<typename Func>
```

Comment:
do we want this in the public API?

Reply:
yes, this is helpful, very common pattern